percona/controller/pgcluster/controller.go

```diff
@@ -310,3 +316,25 @@ func (r *PGClusterReconciler) handleMonitorUserPassChange(ctx context.Context, c
 
 	return nil
 }
+
+func (r *PGClusterReconciler) reconcileCustomExtensions(ctx context.Context, cr *v2.PerconaPGCluster) error {
+	if cr.Spec.CustomExtensions.Storage.Secret == nil {
```

Comment:
Should you have more checking here, like to see if `CustomExtensions` is nil or not and other fields under it?

Besides that, if `Secret` is nil here, should that be an error for the user, to see that Secrets are missing?

Reply:
`CustomExtensions` can't be nil. I added this check to decide if we need to reconcile custom extensions or not. I can't use `extensions` slice since if it's empty, we need to uninstall installed extensions. Which checks can we add?

Reply:
Aha, I see, you have them as value types. I guess it is ok then. Btw since they are value types, I think you can omit `omitempty` from struct tags.